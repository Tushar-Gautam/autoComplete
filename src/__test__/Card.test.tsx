import { render, screen } from "@testing-library/react";
import { Card } from "@components";

describe("Card Component", () => {
  const mockProps = {
    id: 1,
    title: "Sample Book",
    summary: "This is a sample summary.",
    author: "John Doe",
  };

  it("renders the Card component", () => {
    render(<Card {...mockProps} />);
    const cardElement = screen.getByText("Sample Book");
    expect(cardElement).toBeInTheDocument();
  });

  it("displays the correct title, summary, and author", () => {
    render(<Card {...mockProps} />);
    expect(screen.getByText("Sample Book")).toBeInTheDocument();
    expect(screen.getByText("This is a sample summary.")).toBeInTheDocument();
    expect(screen.getByText(/John Doe/i)).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Card {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
