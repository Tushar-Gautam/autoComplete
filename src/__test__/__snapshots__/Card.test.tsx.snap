// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <h3>
      Sample Book
    </h3>
    <p>
      This is a sample summary.
    </p>
    <p>
      <strong>
        Author:
      </strong>
       John Doe
    </p>
  </div>
</DocumentFragment>
`;
